src/Middleware/RateLimiting/test/RateLimitingMiddlewareTests.cs

```diff
@@ -0,0 +1,110 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Testing;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+
+namespace Microsoft.AspNetCore.RateLimiting;
+
+public class RateLimitingMiddlewareTests : LoggedTest
+{
+    [Fact]
+    public void Ctor_ThrowsExceptionsWhenNullArgs()
+    {
+        var options = CreateOptionsAccessor();
+        options.Value.Limiter = new TestPartitionedRateLimiter<HttpContext>();
+
+        Assert.Throws<ArgumentNullException>(() => new RateLimitingMiddleware(
+            null,
+            new NullLoggerFactory().CreateLogger<RateLimitingMiddleware>(),
+            options));
+
+        Assert.Throws<ArgumentNullException>(() => new RateLimitingMiddleware(c =>
+        {
+            return Task.CompletedTask;
+        },
+        null,
+        options));
+    }
+
+    [Fact]
+    public async Task RequestsCallNextIfAccepted()
```

Comment:
Do we have any tests for leases that are acquired or not asynchronously? It'd be nice to have coverage verifying we're properly flowing `HttpContext.RequestAborted` to `WaitAsync`. I guess that could also be verified inline too.

Reply:
> It'd be nice to have coverage verifying we're properly flowing HttpContext.RequestAborted to WaitAsync

isn't that testing the runtime behavior (i.e. it'd wind up being a test of whether or not my TestRateLimiter was implemented correctly)?

Reply:
The bug that we'd want to catch is if we didn't flow `RequestAborted ` through to the call to `WaitAsync`. That would be a bug in the middleware. If the limiter doesn't respect it, that would be a bug in the limiter.

You could just set `HttpContext.RequestAborted = new CancellationToken(true);`, update `TestRateLimiter` to call `token.ThrowIfCanceled()` or whatever, and then verify invoking the middleware throws an `OCE` in a test where it would otherwise fail to acquire the lease gracefully if not for the canceled token.

Reply:
Added